pkg/kubectl/service.go

```diff
@@ -183,8 +183,16 @@ func generate(genericParams map[string]interface{}) (runtime.Object, error) {
 	if len(params["type"]) != 0 {
 		service.Spec.Type = api.ServiceType(params["type"])
 	}
-	if service.Spec.Type == api.ServiceTypeLoadBalancer {
+	switch service.Spec.Type {
+	case api.ServiceTypeNodePort:
+		for i, port := range service.Spec.Ports {
+			service.Spec.Ports[i].NodePort = port.Port
+		}
+	case api.ServiceTypeLoadBalancer:
 		service.Spec.LoadBalancerIP = params["load-balancer-ip"]
+	case "Headless":
```

Comment:
Once upon a time Service was The Way to manage IP addresses.  It turns out we have a lot of kinds of IP addresses and we probably need a more robust API for that.  We don't have that API yet, so many of the historical decisions must  stand.  Ergo, a plethora of options.

Currently "expose" maps pretty closely to the REST representation of a service.  I'm reluctant to depart from that just for one case.  If we need a `--node-port` flag, I can live with that, though node ports were never intended to be pets.  They were really designed to be useful in creating higher-level load-balancers (e.g. ELB).

What subcommands do you think would make this more usable?